### DISCUSSION TITLE
精读React hooks（二）：全面掌握useReducer | J实验室

### DISCUSSION THREAD

**Discussion by giscus** [weijunext/new-blog#8] [General]
# 精读React hooks（二）：全面掌握useReducer | J实验室

全栈工程师，Next.js开源手艺人，掘金签约作者，折腾才有未来

https://weijunext.com/article/react-hook-useReducer

**Comment by s0974092** [comment]
感謝您詳細說明useReducer的參數用法，我把您的代碼實際跑在React上，發現第二個方法中的 getInitialCount 函數返回值，應該為 { counter: number }，才不會發生NaN的問題。

function getInitialCount() {
  const savedCount = localStorage.getItem("count");
  return savedCount ? Number(savedCount) : 0; // 這個應該為 return { count: _.isNumber(savedCount) ? Number(savedCount) : 0 }; 否則，取不到 count的值。
}
